chia/_tests/wallet/test_revcat_lifecycle.py

```diff
@@ -0,0 +1,667 @@
+    extra_deltas: Optional[list[int]] = None,
+    additional_spends: list[WalletSpendBundle] = [],
+    limitations_solutions: Optional[list[Program]] = None,
+    cost_logger: Optional[CostLogger] = None,
+    cost_log_msg: str = "",
+) -> int:
+    if limitations_solutions is None:
+        limitations_solutions = [Program.to([])] * len(coins)
+    if extra_deltas is None:
+        extra_deltas = [0] * len(coins)
+
+    spendable_cat_list: list[SpendableCAT] = []
+    for coin, innersol, proof, limitations_solution, extra_delta in zip(
+        coins, inner_solutions, lineage_proofs, limitations_solutions, extra_deltas
+    ):
+        spendable_cat_list.append(
+            SpendableCAT(
+                coin,
+                tail.get_tree_hash(),
+                p2_puzzle,
+                innersol,
+                limitations_solution=limitations_solution,
+                lineage_proof=proof,
+                extra_delta=extra_delta,
+                limitations_program_reveal=(
+                    tail if reveal_limitations_program else Program.to([])
+                ),
+            )
+        )
+
+    spend_bundle = unsigned_spend_bundle_for_spendable_cats(CAT_MOD, spendable_cat_list)
+    agg_sig = AugSchemeMPL.aggregate(signatures)
+    final_bundle = WalletSpendBundle.aggregate(
+        [
+            *additional_spends,
+            spend_bundle,
+            WalletSpendBundle([], agg_sig),
+        ]  # "Signing" the spend bundle
+    )
+    for spend in spend_bundle.coin_spends:
+        print(spend.coin.name().hex())
+    if cost_logger is not None:
+        final_bundle = cost_logger.add_cost(cost_log_msg, final_bundle)
+
+    result = await sim_client.push_tx(final_bundle)
+    assert result == expected_result
+    cost = cost_of_spend_bundle(spend_bundle)
+    await sim.farm_block()
+    return cost
+
+
+# Helper function
+async def make_and_spend_bundle(
+    sim: SpendSim,
+    sim_client: SimClient,
+    coin: Coin,
+    delegated_puzzle: Program,
+    coinsols: list[CoinSpend],
+    ex_error: Optional[Err] = None,
+    fail_msg: str = "",
+    cost_logger: Optional[CostLogger] = None,
+    cost_log_msg: str = "",
+):
+    signature: G2Element = sign_delegated_puz(delegated_puzzle, coin)
+    spend_bundle = SpendBundle(
+        coinsols,
+        signature,
+    )
+    if cost_logger is not None:
+        spend_bundle = cost_logger.add_cost(cost_log_msg, spend_bundle)
+
+    try:
+        for spend in spend_bundle.coin_spends:
+            print(f"SPEND: {spend.coin.name().hex()}")
+        _result, error = await sim_client.push_tx(spend_bundle)
+        if error is None:
+            await sim.farm_block()
+        elif ex_error is not None:
+            assert error == ex_error
+        else:
+            raise TransactionPushError(error)
+    except AssertionError:
+        raise AssertionError(fail_msg)
+
+
+@pytest.mark.anyio
+async def test_revocable_cat_lifecycle(cost_logger):
```

Comment:
I realize this is copied from a real in-"production" test but it's from like 3 years ago.  Using the standard puzzle as the p2 puzzle really over complicates the whole file and makes it difficult to review and get accurate cost metrics.  I would rather use `Program.to(1)` as the inner puzzle where possible because it's the minimal-est puzzle that allows for what you need and therefore weighs on the cost number the least.

Reply:
That makes sense, will use that!